language/tools/move-cli/src/main.rs

```diff
@@ -170,6 +176,23 @@ impl Move {
         self.mode.compiled_modules(&self.get_package_dir())
     }
 
+    /// Return `true` if `path` is a Move bytecode file based on its extension
+    fn is_bytecode_file(path: &Path) -> bool {
+        path.extension()
+            .map_or(false, |ext| ext == MOVE_COMPILED_EXTENSION)
+    }
+
+    /// Return `true` if path contains a valid Move bytecode module
+    fn contains_module(path: &Path) -> bool {
+        Self::is_bytecode_file(path) && {
+            if let Ok(bytes) = fs::read(path) {
+                CompiledModule::deserialize(&bytes).is_ok()
+            } else {
```

Comment:
Not a huge fan of if-let-else. Especially when there are only two cases...

Reply:
I would actually complain more about if-let-else for larger variants. IMO, the value of insisting on `match` over `if let` is that it keeps you from ignoring cases. But when there are only two cases and one of them is `None` or `Err`, using `if let` seems fairly reasonable (unless I'm missing something).